code/modules/food_and_drinks/food.dm

```diff
@@ -47,16 +47,17 @@
 	..()
 
 /obj/item/reagent_containers/food/proc/check_for_ants()
+	last_ant_time = world.time
 	var/turf/T = get_turf(src)
-	if(isturf(loc) && !locate(/obj/structure/table) in T)
-		if(ant_location == T)
-			if(prob(15))
-				if(!locate(/obj/effect/decal/cleanable/ants) in T)
-					new /obj/effect/decal/cleanable/ants(T)
-					antable = FALSE
-					desc += " It appears to be infested with ants. Yuck!"
-					reagents.add_reagent("ants", 1) // Don't eat things with ants in it you weirdo.
-		else
-			ant_location = T
+	if((locate(/obj/structure/table) in T) || (locate(/obj/structure/rack) in T) || !isturf(loc))
```

Comment:
can you do the `!isturf()` check first just to save on searching the turf for tables/racks